### DISCUSSION TITLE
Remaining days not checked for wildcard certificates?

### DISCUSSION THREAD

**Discussion by ahbnr** [go-acme/lego#2300] [Q&A] [answered]
Hi!

When my `lego` job is called for renewal, I get the following message for one of my domains:
`The certificate expires in 84 days, the number of days defined to perform the renewal is 30: no renewal.`
This is exactly what I want, and it seems to work correctly.

However, for my wildcard certificates, I do not get this message. Instead I get
```
lego-1  | 2024/10/12 12:51:04 [INFO] [*.my.domain] acme: Obtaining bundled SAN certificate
lego-1  | 2024/10/12 12:51:05 [INFO] [*.my.domain] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/...
lego-1  | 2024/10/12 12:51:05 [INFO] [*.my.domain] acme: authorization already valid; skipping challenge
lego-1  | 2024/10/12 12:51:05 [INFO] [*.my.domain] acme: Validations succeeded; requesting certificates
lego-1  | 2024/10/12 12:51:17 [INFO] [*.my.domain] Server responded with a certificate.
```

I.e. even though the certificate is new and has many days left, it still requests it from the server.
**Is this a bug, or intended behaviour?**

Thank you!

**Comment by ldez** [comment]
Hello,

there is no special process for wildcards: the same expiration detection is applied.

To diagnose, I need:
- the command(s) used to create the certificates
- the command(s) used to renew

**Reply by ahbnr** [reply, marked as answer]
Hi ldez,

wow you are fast! Thank you very much for the quick response!

> To diagnose, I need:
>
>   the command(s) used to create the certificates
>   the command(s) used to renew

Your questions were on-point. There was a bug in *my* invocation logic, and it called `run` instead of `renew` for the wildcard certificates.
I'm sorry I have wasted your time!

**Reply by ahbnr** [reply]
The bug occurred because my scipt does a manual check for whether the certificate file already exists. If it does, I execute `renew` instead of `run`.
Here, I made the mistake of not replacing `*` with `_` when looking for the certificate file.

Hence, a follow-up question:
Can lego do this automatically? I.e. only execute the `run` procedures if there is no existing data for a certificate, and automatically use `renew` otherwise?
Perhaps I missed it, but I didn't see any such option.

**Reply by ldez** [reply]
Currently, lego doesn't support that but we are issues on the topic: #216, #290, #693

**Reply by ahbnr** [reply]
Alright! Again, thank you a lot for your help, and for your work on this great tool!

**Reply by ldez** [reply]
I feel you enjoyed my work, can you consider donating, this will be appreciated, thank you :heart:

https://github.com/sponsors/ldez

**Reply by ahbnr** [reply]
I sent you a small token of appreciation :)

I'm not using this for work but I'm just tinkering around with it for my homelab. Thus, I unfortunately can not sponsor you on a monthly basis.
